src/Functions/FunctionsHashing.h

```diff
@@ -1201,6 +1207,74 @@ class FunctionAnyHash : public IFunction
                     column->getName(), getName());
     }
 
+    template <bool first>
+    void executeNothing(const KeyColumnsType &, const IColumn *, typename ColumnVector<ToType>::Container & vec_to) const
+    {
+        /// This value shouldn't affect anything, it should only appear inside null Nullable or
+        /// empty Array, where the caller will ignore this and assign their own hash value.
+        /// Fill it with zeroes just in case, to avoid leaking memory contents if something's broken.
+        vec_to.assign(vec_to.size(), ToType(0));
+    }
+
+    template <bool first>
+    void executeNullable(const KeyColumnsType & key_cols, const IDataType * from_type, const IColumn * column, typename ColumnVector<ToType>::Container & vec_to) const
+    {
+        if (const auto * col_from = checkAndGetColumn<ColumnNullable>(column))
+        {
+            const auto * nested_type = assert_cast<const DataTypeNullable &>(*from_type).getNestedType().get();
+            const auto & nested_col = col_from->getNestedColumn();
+
+            KeyType key {};
+            ToType null_hash;
+            if constexpr (Keyed)
+            {
+                /// Make the hash depend on key.
```

Comment:
Btw, I noticed two existing bugs: `sipHash128([])` has only lower 32 nonzero bits, and any keyed hash of `[]` doesn't depend on key:
```sql
SELECT reinterpret(sipHash128Keyed((number, number), []), 'UInt128')
FROM numbers(2)

Query id: 4a7f8a7f-c98b-4f41-897d-55d773e67fea

   ┌─reinterpret(⋯ 'UInt128')─┐
1. │               4249604106 │
2. │               4249604106 │
   └──────────────────────────┘
```
But idk how to fix it without breaking compatibility, guess we should leave it as is.